UnityProject/Assets/Scripts/Managers/ChatRelay.cs

```diff
@@ -71,6 +72,28 @@ public void AddToChatLogClient(string message, ChatChannel channels)
 	private void PropagateChatToClients(ChatEvent chatEvent)
 	{
 		var players = PlayerList.Instance.InGamePlayers;
+
+		//Local chat range checks:
+		if (chatEvent.channels == ChatChannel.Local) {
+			var speaker = PlayerList.Instance.Get(chatEvent.speaker);
+			RaycastHit2D hit;
+			LayerMask layerMask = 1 << 9; //Walls layer
+			for (int i = 0; i < players.Count(); i++){
+				if(Vector2.Distance(speaker.GameObject.transform.position,
+				                    players[i].GameObject.transform.position) > 14f){
+					//Player in the list is too far away for local chat, remove them:
+					players.Remove(players[i]);
+				} else {
+					//within range, but check if they are in another room or hiding behind a wall
```

Comment:
could probably use Pulsey's room detection stuff here and remove people whose location's room ids are different to the speaker's

Reply:
but that would mean that people standing on both sides of an open door couldn't talk via local chat.

Reply:
Oh, good point!
I think we should only make them recalculate on wall placement/deconstruction – when we implement building